src/Commands/core/UpdateDBCommands.php

```diff
@@ -357,20 +362,26 @@ public static function updateDoOnePostUpdate($function, DrushBatchContext $conte
      * @param array $results
      * @param array $operations
      */
-    public static function updateFinished($success, $results, $operations)
+    public function updateFinished($success, $results, $operations)
     {
-        // No code needed but the batch result bookkeeping fails without a finished callback.
-        $noop = 1;
+        // Check if updates or post-updates changed the installed extensions.
+        \Drupal::configFactory()->reset('core.extension');
+        $final_extension_list = \Drupal::config('core.extension')->getRawData();
+        if ($final_extension_list !== $this->originalExtensionList) {
+            // Invalidate the container if the installed extension list has been changed.
+            \Drupal::service('kernel')->invalidateContainer();
+        }
     }
 
-
     /**
      * Start the database update batch process.
      * @param $options
      * @return bool
      */
     public function updateBatch($options)
     {
+        $this->originalExtensionList = \Drupal::config('core.extension')->getRawData();
```

Comment:
```suggestion
        // Track the currently installed extensions so that we can detect if any
        // new extensions are added in the update.
        $this->originalExtensionList = \Drupal::config('core.extension')->getRawData();
```